src/mug/synth/synth.cljs

```diff
@@ -0,0 +1,36 @@
+(ns mug.core.synth
+  (:require
+    [cljs-bach.synthesis
+     :as b
+     :refer [connect->]
+     :rename {connect-> <>}]))
+
+(defonce context (b/audio-context))
+
+(defn play
+  ([synth duration]
+   (let [node (<> synth b/destination)
+         at (b/current-time context)]
+     (b/run-with node context at duration)))
+  ([synth duration gain]
+   (play (<> synth (b/gain gain)) duration)))
+
+(defn raw-buffer [generate-bit! context duration sample-rate]
+  (let [frame-count (* sample-rate duration)
+        buffer (.createBuffer context 1 frame-count sample-rate)
+        data (.getChannelData buffer 0)]
+    (doseq [i (range frame-count)] (aset data i (generate-bit! i)))
+    buffer))
+
+(def buffer (memoize raw-buffer))
+
+(defn noise [generate-bit! sample-rate]
+  (fn [context at duration]
+    (b/source
+      (doto (.createBufferSource context)
+        (-> .-buffer (set! (buffer generate-bit! context duration sample-rate)))
+        (.start at)))))
+
+(defn classic [x]
```

Comment:
Looke like this code is unused

Reply:
It's supposed to be used from repl as an argument to `noise`

Reply:
Got it